modules/docs/arrow-docs/docs/docs/arrow/typeclasses/traverse/README.md

```diff
@@ -11,16 +11,491 @@ redirect_from:
 {:.intermediate}
 intermediate
 
-TODO. Meanwhile you can find a short description in the [intro to typeclasses]({{ '/docs/typeclasses/intro/' | relative_url }}).
+In functional programming it is very common to encode "behaviors" as data types - common behaviors include `Option` for possibly missing values, `Either` and `Validated` for possible errors, and [`Promise`]({{ '/docs/effects/promise/' | relative_url }}) for asynchronous computations, which has the same purpose as a `Future`, only pure and lazy.
 
+These behaviors tend to show up in functions working on a single piece of data - for instance parsing a single `String` into an `Int`, validating a login, or asynchronously fetching website information for a user.
+
+```kotlin:ank
+import arrow.core.Either
+import arrow.core.Option
+import arrow.fx.ForIO
+import arrow.fx.Promise
+
+interface Profile
+interface User
+
+fun userInfo(u: User): Promise<ForIO, Profile> = TODO()
```

Comment:
This is just some nitpicking but `userInfo` would be better suitable for a field name. What do you think about `loadUserInfo`, or something similar?